### PR TITLE
[FLINK-11146] Remove invalid codes in ClusterClient

### PR DESCRIPTION
**Pull request by TisonKun**
## What is the purpose of the change

Remove invalid codes in `ClusterClient`.

This pr is aimed at dropping actor/message dependencies in `ClusterClient`. Mainly. it does

1. Delete implementation of following methods and make them `abstract`, because they are implemented by `MiniClusterClient` and `RestClusterClient`, and the implementation in `ClusterClient` depends on legacy logic and thus is invalid. METHODS include 
  - getJobStatus(jobID)
  - cancel(jobID)
  - cancelWithSavepoint(jobID, savepointDir)
  - stop(jobID)
  - triggerSavepoint(jobID, savepointDir)
  - disposeSavepoint(savepointPath)
  - listJobs()
  - getAccumulators(jobID, ClassLoader) 

2. Delete several legacy abstract methods. Since the implementations are simply return a literal None value. Also delete their use points, which includes
  - logAndSysout(String message)
  - waitForClusterToBeReady() 
  - getClusterStatus()
  - getNewMessages()
  - getMaxSlots()
  - hasUserJarsInClassPath(List<URL> userJarFiles)

~~3. There is still one method depend on legacy mode but I am not sure if we can directly remove it or should be there a port. It is `endSession(JobID jobID)`. I can see now the message(`JobManagerMessages.RemoveCachedJob`) it sends is what no one processes and thus guess that we can directly remove this method as well as its use point in `ContextEnvironment` and `RemoteExecutor`. But I think it is necessary to involve a committer who know how this logic works to confirm that we can do that.~~

3. Delete usages of `getJobManagerGateway` and `LazyActorSystemLoader`, including drop method `endSession`.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector:(no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

cc @tillrohrmann @mxm

### CODE REVIEW

**Comment by mxm** [comment]
>    1. There is still one method depend on legacy mode but I am not sure if we can directly remove it or should be there a port. It is `endSession(JobID jobID)`. I can see now the message(`JobManagerMessages.RemoveCachedJob`) it sends is what no one processes and thus guess that we can directly remove this method as well as its use point in `ContextEnvironment` and `RemoteExecutor`. But I think it is necessary to involve a committer who know how this logic works to confirm that we can do that.

That batch-related feature was never fully implemented. Seems fair to remove it.

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/util/FakeClusterClient.java [hunk 1]
```diff
@@ -1,79 +0,0 @@
...
-
-import org.apache.hadoop.yarn.api.records.ApplicationId;
-
-import java.net.URL;
-import java.util.List;
-
-/**
- * Dummy {@link ClusterClient} for testing purposes (extend as needed).
- */
-public class FakeClusterClient extends ClusterClient<ApplicationId> {
```
Comment (zentol) [inline flink-yarn-tests/src/test/java/org/apache/flink/yarn/util/FakeClusterClient.java]: it should be perfectly fine to retain this one.

Comment (TisonKun) [reply flink-yarn-tests/src/test/java/org/apache/flink/yarn/util/FakeClusterClient.java]: Right. I'd adjust this dummy implementation.

File: flink-clients/src/main/java/org/apache/flink/client/RemoteExecutor.java [hunk 2]
```diff
@@ -226,32 +226,6 @@ public String getOptimizerPlanAsJSON(Plan plan) throws Exception {
 
 	@Override
 	public void endSession(JobID jobID) throws Exception {
-		if (jobID == null) {
```
Comment (TisonKun) [inline flink-clients/src/main/java/org/apache/flink/client/RemoteExecutor.java]: (1/2) Remove `endSession(JobID)` use point. Please check that this doesn't change current behaviors.

File: flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java [hunk 3]
```diff
@@ -74,7 +74,6 @@ public String getExecutionPlan() throws Exception {
 
 	@Override
 	public void startNewSession() throws Exception {
-		client.endSession(jobID);
```
Comment (TisonKun) [inline flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java]: (2/2) Remove `endSession(JobID)` use point. Please check that this doesn't change current behaviors.

**Comment by TisonKun** [comment]
@mxm do you know if there is a committer could help review these changes? It stalls for several days but the changes look not quite complex.

**Comment by mxm** [comment]
I'd merge. Anything blocking this from being merged?

**Comment by TisonKun** [comment]
@mxm No from my side if you can double check the removals of `endSession(JobID)` are safe.

**Pull request merged**